### ISSUE TITLE
Server side: ReferenceError when using component with namespace

### ISSUE THREAD

**Issue by snokleby** [reactjs/React.NET#260]
I have a React-component that uses require() , so I use Browserify (in grunt) to make this accessible to the browser, and ReactJS.net server-side. It works when I use "manual" client-side rendering, but I get the following error when trying to render it server-side:

`ReferenceError: OurReactComponents is not defined
   at Script Document [8] [temp]:1:8 -> typeof OurReactComponents.LoanJumboTron !== 'undefined'
Line: 0
Column:0`

I'm pretty new to CommonJS-modules and Browserify, so this might be overly complex:

In ReactConfig.cs:
````
ReactSiteConfiguration.Configuration = new ReactSiteConfiguration()
                .SetLoadBabel(false)
                .SetLoadReact(false)               
                .AddScriptWithoutTransform("~/GeneratedScripts/React.js")
                .AddScriptWithoutTransform("~/GeneratedScripts/ReactStandalone.js")
                .AddScriptWithoutTransform("~/GeneratedScripts/ReactServerStandalone.js")
                .AddScriptWithoutTransform("~/GeneratedScripts/ReactDOMStandAlone.js")
                .AddScriptWithoutTransform("~/GeneratedScripts/JumboTron.js");

            ReactSiteConfiguration.Configuration.AllowMsieEngine = false;
````

The three standalone-files are for exposing React, ReactDOM and ReactDOMServer as global object, like this (Gruntfile.js):
```
reactStandAlone: {
                src: [
                    "scripts/jsx/Loaders/React.js"
                ],
                dest: 'GeneratedScripts/ReactStandalone.js',
                options: {
                    browserifyOptions: {
                        standalone: 'React'
                    },
                    external: ['react', 'react-dom']
                }
            },
```
Loaders/React.js looks like this:
`module.exports = require('react');`
(This might be where I'm showing that I have no idea what I'm doing)

The file with our own code is built like this:
```
jumboTron: {
                src: [
                    "scripts/jsx/JumboTronLoader.js"
                ],
                dest: 'GeneratedScripts/JumboTron.js',
                options: {
                    alias: ['react:', 'react-dom:'],
                    browserifyOptions: {
                        standalone: 'OurReactComponents',
                        debug: false
                    },
                    external: [
                      'react', 'react-dom'
                    ]
                }
            },
```
... and JumboTronLoader.js 
```
module.exports = {
    LoanJumboTron: require('./LoanJumboTron.jsx')
};
```
Finally, I create the bundle containing react:
```
react: {
                src: [
                    "scripts/jsx/Loaders/React.js"
                ],
                dest: 'GeneratedScripts/React.js',
                options: {
                    alias: ['react:', 'react-dom:'],
                    browserifyOptions: {
                        debug: false
                    },
                    external: []
                }
            },
```
And Loaders/React.js looks like this: 
`module.exports = require('react');`

A lot more plumbing than I would have liked, and could hopefully have been made simpler. 
But in my mind (and Chrome) it should work. I'm making the required objects available globally, and the react-component gets its needed dependencies.

OurReactComponents looks like this:
![reactcomponent](https://cloud.githubusercontent.com/assets/3669251/14891669/d903d1b8-0d67-11e6-8192-c68c2dc5fbae.png)

What am I doing wrong? any help is greatly appreciated.

**Comment by snokleby** [comment]
I forgot to include the .cshtml part:
```
 @Html.React("OurReactComponents.LoanJumboTron", new
    { //...}
```

**Comment by Daniel15** [comment]
Hmmm, interesting. What does `GeneratedScripts/JumboTron.js` look like? I wonder if it's exposing the globals in a way that works in a browser but not in server-side rendering.

**Comment by snokleby** [comment]
Thanks for helping me figure this out. This should be solved by #267 .
